### PR TITLE
Remove handling of multiple timesteps from ensemble calibration

### PR DESCRIPTION
**Pull request by gavinevans**
Addresses: Part of #854 

Description
Remove handling of multiple timesteps from ensemble calibration, so that the unit tests are simplified and so as to simplify further changes. This functionality is redundant, as given the IMPROVER framework, the ensemble calibration will only ever to provided with a file that contains a single timestep.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: lib/improver/ensemble_calibration/ensemble_calibration.py [hunk 1]
```diff
@@ -853,140 +766,93 @@ def _apply_params(
 
         Returns:
             (tuple) : tuple containing:
-                **calibrated_forecast_predictor_all_dates** (CubeList):
-                    List of cubes containing the calibrated forecast predictor.
-                **calibrated_forecast_var_all_dates** (CubeList):
-                    List of cubes containing the calibrated forecast variance.
-                **calibrated_forecast_coefficients_all_dates** (CubeList):
+               **calibrated_forecast_predictor** (iris.cube.Cube):
```
[the remark this answers is not in the document]
Comment (gavinevans) [reply lib/improver/ensemble_calibration/ensemble_calibration.py]: I've updated the Args section. You're correct that the `optimised_coeffs` is a dict type.

File: lib/improver/ensemble_calibration/ensemble_calibration.py [hunk 2]
```diff
@@ -853,140 +766,93 @@ def _apply_params(
...
+        elif predictor_of_mean_flag.lower() in ["realizations"]:
+            # Calculate predicted mean = a + b*X, where X is the
+            # raw ensemble mean. In this case, b = beta^2.
+            beta_values = np.array([], dtype=np.float32)
+            for key in optimised_coeffs_at_date.keys():
+                if key.startswith("beta"):
+                    beta_values = (
+                        np.append(beta_values, optimised_coeffs_at_date[key]))
+            beta = np.concatenate(
+                [[optimised_coeffs_at_date["a"]], beta_values**2])
```
[the remark this answers is not in the document]
Comment (gavinevans) [reply lib/improver/ensemble_calibration/ensemble_calibration.py]: I've updated these "a" values to be "alpha" values in #850, so this should be updated once #850 is merged and I update this branch.

File: lib/improver/tests/ensemble_calibration/ensemble_calibration/test_ApplyCoefficientsForEnsembleCalibration.py [hunk 3]
```diff
@@ -794,105 +694,25 @@ def test_too_many_coefficients(self):
...
-        cube = concatenate_cubes(CubeList([cube1, cube2]))
 
         optimised_coeffs = {}
-
-        for time_slice in cube.slices_over("time"):
-            the_date = datetime_from_timestamp(time_slice.coord("time").points)
-            optimised_coeffs[the_date] = self.default_optimised_coeffs
+        the_date = datetime_from_timestamp(cube.coord("time").points)
+        optimised_coeffs[the_date] = np.array(
+            [5, 1, 0, 0.57, 0.6, 0.6], dtype=np.float32)
```
[the remark this answers is not in the document]
Comment (gavinevans) [reply lib/improver/tests/ensemble_calibration/ensemble_calibration/test_ApplyCoefficientsForEnsembleCalibration.py]: I've corrected this.

File: lib/improver/ensemble_calibration/ensemble_calibration.py [hunk 4]
```diff
@@ -907,156 +820,109 @@ def _apply_params(
...
-                    predicted_mean = np.dot(all_data, beta)
-                    calibrated_forecast_predictor_at_date = (
-                        forecast_predictor_at_date)
-                elif predictor_of_mean_flag.lower() in ["realizations"]:
-                    # Calculate predicted mean = a + b*X, where X is the
-                    # raw ensemble mean. In this case, b = beta^2.
-                    beta = np.concatenate(
-                        [[optimised_coeffs_at_date["alpha"]],
-                         optimised_coeffs_at_date["beta"]**2])
-                    forecast_predictor = (
```
Comment (fionaRust) [inline lib/improver/ensemble_calibration/ensemble_calibration.py]: Why have you got rid of this line?

Comment (gavinevans) [reply lib/improver/ensemble_calibration/ensemble_calibration.py]: This line was not having any effect as `forecast_predictor` is not used again in this function.

**Comment by fionaRust** [comment]
Thank you for doing this separately, it was a lot easier to review.

**Pull request merged**